src/Messaging/src/RabbitMQ/Core/BatchingRabbitTemplate.cs

```diff
@@ -93,7 +93,7 @@ public override void Send(string exchange, string routingKey, IMessage message,
                 {
                     _cancellationTokenSource = new CancellationTokenSource();
                     TimeSpan delay = next.Value - DateTime.Now;
-                    _scheduledTask = Task.Run(() => ReleaseBatches(delay, _cancellationTokenSource.Token), _cancellationTokenSource.Token);
+                    _scheduledTask = Task.Run(() => ReleaseBatchesAsync(delay, _cancellationTokenSource.Token), _cancellationTokenSource.Token);
```

Comment:
Should there also be an `await` on this line? cc @dtillman @hananiel

Reply:
Yes, I suspect it should. The `ReleaseBatchesAsync` method contains await points. So any exception thrown after the first await point goes unnoticed. Maybe even crash the process, although I'm not entirely sure about that as it depends on the execution context.

Reply:
This concern will be revisited in another issue that I'll create momentarily

Reply:
Resolving, this is trakced at #983.